aws_lambda_powertools/metrics/provider/datadog/datadog.py

```diff
@@ -0,0 +1,291 @@
+from __future__ import annotations
+
+import json
+import logging
+import numbers
+import os
+import time
+import warnings
+from typing import Any, Callable, Dict, List, Optional
+
+from aws_lambda_powertools.metrics.exceptions import MetricValueError, SchemaValidationError
+from aws_lambda_powertools.metrics.provider import BaseProvider
+from aws_lambda_powertools.shared import constants
+from aws_lambda_powertools.shared.functions import resolve_env_var_choice
+from aws_lambda_powertools.utilities.typing import LambdaContext
+
+logger = logging.getLogger(__name__)
+
+# Check if using datadog layer
+try:
+    from datadog_lambda.metric import lambda_metric  # type: ignore
+except ImportError:  # pragma: no cover
+    lambda_metric = None  # pragma: no cover
```

Comment:
You can test this by patching the Import system

```python
import sys


def test_datadog_module_not_found(monkeypatch):
    monkeypatch.setitem(sys.modules, 'datadog_lambda', None)
    ...
```

None here would generate a ImportError if I recall correctly

Reply:
Hmmmm not exactly. I can't access the loaded modules inside the test. I will continue to investigate.